crates/pathfinder/src/rpc/api.rs

```diff
@@ -950,6 +953,52 @@ impl RpcApi {
             // flatten is unstable
             .and_then(|x| x)
     }
+
+    /// Submit a new transaction to be added to the chain.
+    ///
+    /// This method just forwards the request received over the JSON-RPC
+    /// interface to the sequencer.
+    pub async fn add_invoke_transaction(
+        &self,
+        call: Call,
+        signature: Vec<CallSignatureElem>,
+        max_fee: Fee,
+        version: TransactionVersion,
+    ) -> RpcResult<InvokeTransactionResult> {
+        let mut call: sequencer::request::Call = call.into();
+        call.signature = signature;
+
+        let result = self
+            .sequencer
+            .add_invoke_transaction(call, max_fee, version)
+            .await?;
+        Ok(InvokeTransactionResult {
+            txn_hash: result.transaction_hash,
+        })
+    }
+
+    /// Submit a new deploy contract transaction.
+    ///
+    /// This method just forwards the request received over the JSON-RPC
+    /// interface to the sequencer.
+    pub async fn add_deploy_transaction(
+        &self,
+        contract_address_salt: ContractAddressSalt,
+        constructor_calldata: Vec<ConstructorParam>,
+        contract_definition: ContractDefinition,
+    ) -> RpcResult<DeployTransactionResult> {
+        let result = self
+            .sequencer
+            .add_deploy_transaction(
+                contract_address_salt,
+                constructor_calldata,
+                contract_definition,
+            )
+            .await?;
+        Ok(DeployTransactionResult {
+            txn_hash: result.transaction_hash,
```

Comment:
Around here will be the
```suggestion
            txn_hash: result.transaction_hash,
            // TODO: contract address when specs include it
```

Or would it be somewhere else?

Reply:
I wonder if I did misunderstand when I understood that the response would be carrying the address but it isn't, and has deny_extra_fields..

Reply:
I've added the `contract_address` field. (And `txn_hash` has been renamed to `transaction_hash` in the spec, so fixed that too.)